### DISCUSSION TITLE
Logging out?

### DISCUSSION THREAD

**Discussion by mhpaler** [vikaspotluri123/discourse-on-ghost#81] [General]
Presently, you need to log out of both platforms manually in order to fully log out. I think it would be nice, if one logs out on _either_ Ghost or DIscourse, that you are logged out of both platforms. Any ideas on whether this can be achieved with configuration alone?

**Comment by vikaspotluri123** [comment]
Functionally, I fully agree!

Technically, it seems a lot more complicated 😄

Thinking out loud:

 - On the Ghost side, there needs to be some way to detect when a user logs out using Portal.
 - It seems that Discourse has an [API to log users out](https://meta.discourse.org/t/setup-discourseconnect-official-single-sign-on-for-discourse-sso/13045#logging-off-users-13), but it will remove all of the user's sessions, not just the one associated with their current device/browser. Since Discourse and Ghost are completely isolated from that perspective, I don't think it'd be possible to link the sessions together
 - It doesn't seem possible to host a static page on Discourse - if it was, after detecting the logout action in Ghost, we could redirect to a special static page
 - However, Ghost does support static pages - so if a user logs out of Discourse, **and** we can get notified client-side, logging someone out of Ghost would be easy
 - I've curious to know how many people manually log out of sites - this might be a tedious issue with minimal benefit 😬

**Reply by mhpaler** [reply]
This is tough one. Agree with you that it is probably too much work to support an infrequent/rare use case. That said, IMHO, it's not good UX to be variously logged in to a platform and my gut tells me its a doable problem.

Given this, perhaps it's best to just backlog it as a feature for now. It will only take one person to put the time in figuring it out, and then everyone benefits!
